experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/CompositionLocals.kt

```diff
@@ -0,0 +1,8 @@
+package example.imageviewer.view
+
+import androidx.compose.runtime.ProvidableCompositionLocal
+import androidx.compose.runtime.staticCompositionLocalOf
+import example.imageviewer.Dependencies
+
+internal val LocalDependencies: ProvidableCompositionLocal<Dependencies> =
+    staticCompositionLocalOf { error("Dependencies not initialized") }
```

Comment:
It is an interesting suggestion, but I think this change should be made in another PR.
We can apply this logic by ourselves in another branch with big changes.